File: maze/tests/fixtures/sample.ts
// Sample TypeScript code for constraint extraction testing
interface UserAuth {
    username: string;
    password: string;
    email?: string;
}

class AuthService {
    private users: Map<string, UserAuth> = new Map();
    
    async authenticate(username: string, password: string): Promise<boolean> {
        const user = this.users.get(username);
        if (!user) {
            return false;
        }
        return user.password === password;
    }
    
    register(user: UserAuth): void {
        if (this.users.has(user.username)) {
            throw new Error("User already exists");
        }
        this.users.set(user.username, user);
    }
}

File: maze/tests/fixtures/sample.py
# Sample Python code for constraint extraction testing
from typing import Optional, Dict
from dataclasses import dataclass

@dataclass
class UserAuth:
    username: str
    password: str
    email: Optional[str] = None

class AuthService:
    def __init__(self):
        self.users: Dict[str, UserAuth] = {}
    
    async def authenticate(self, username: str, password: str) -> bool:
        """Authenticate a user with username and password."""
        user = self.users.get(username)
        if not user:
            return False
        return user.password == password
    
    def register(self, user: UserAuth) -> None:
        """Register a new user."""
        if user.username in self.users:
            raise ValueError("User already exists")
        self.users[user.username] = user
